community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/codegen/ir/aggregation/AggregateExpression.scala

```diff
@@ -48,12 +48,17 @@ abstract class BaseAggregateExpression(expression: CodeGenExpression, distinct:
     expression match {
       case NodeExpression(v) => primitiveIfNot(v, structure)(block(_))
       case RelationshipExpression(v) => primitiveIfNot(v, structure)(block(_))
-      case _ =>
+      case expr =>
         val tmpName = context.namer.newVarName()
         structure.assign(tmpName, expression.codeGenType, expression.generateExpression(structure))
-        structure.ifNonNullStatement(structure.loadVariable(tmpName)) { body =>
+        val perhapsCheckForNotNullStatement: ((MethodStructure[E]) => Unit) => Unit = if (expr.nullable)
```

Comment:
Not very important, but the general convention seem to be to use "maybe" instead of "perhaps" in naming.